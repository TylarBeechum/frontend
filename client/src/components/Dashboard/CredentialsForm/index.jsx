import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { MaskedInput, Select, Box, Heading, TextArea } from 'grommet';

import {
  BaseForm,
  BaseTextInput,
  BaseFormField,
  BaseButton
} from '../../../styles/themes';

import queries from './queries';
import Field from '../../Field';

const CredentialsForm = ({ history }) => {
  const [credsInputs, setCredsInputs] = useState({
    name: '',
    description: '',
    type: '',
    studentEmail: '',
    imageUrl: '',
    criteria: '',
    issuedOn: '',
    expirationDate: '',
    schoolId: localStorage.id
  });

  const handleChanges = e => {
    setCredsInputs({
      ...credsInputs,
      [e.target.name]: e.target.value
    });
  };

  const handleSubmit = async e => {
    e.preventDefault();
    try {
      history.push('/dashboard');
      await queries.addNewCredentials(credsInputs);
    } catch (error) {
      console.error(error);
    }
  };

  return (
    <Container>
      <section>
        <h2>Issue Credential</h2>
        <BaseForm onSubmit={handleSubmit}>
          <Box>
            <BaseFormField label="Name of Student">
              <BaseTextInput
                name="ownerName"
                placeholder="Jane Doe"
                onChange={handleChanges}
                value={credsInputs.ownerName}
                required
              />
            </BaseFormField>
            <BaseFormField label="Name of Student">
              <BaseTextInput
                name="credName"
                placeholder="Masters in Philopsophy"
                onChange={handleChanges}
                value={credsInputs.credName}
                required
              />
            </BaseFormField>
            <BaseFormField label="Type">
              <BaseTextInput
                name="type"
                placeholder="Masters, PhD, Cert, etc."
                onChange={handleChanges}
                value={credsInputs.type}
                required
              />
            </BaseFormField>
            <BaseFormField label="Description">
              <TextArea
                name="description"
                placeholder="Summary of credential"
                onChange={handleChanges}
                value={credsInputs.description}
                required
              />
            </BaseFormField>

            <BaseFormField label="Student Email">
              <BaseTextInput
                name="studentEmail"
                placeholder="Jane.Doe@gmail.com"
                onChange={handleChanges}
                value={credsInputs.studentEmail}
                required
              />
            </BaseFormField>
            <BaseFormField label="School Seal Image URL">
              <BaseTextInput
                name="imageUrl"
                placeholder="Image"
                onChange={handleChanges}
                value={credsInputs.imageUrl}
                required
              />
            </BaseFormField>
            <BaseFormField label="Criteria">
              <BaseTextInput
                name="criteria"
                placeholder="Enter the criteria for the credentials"
                onChange={handleChanges}
                value={credsInputs.criteria}
                required
              />
            </BaseFormField>
            <BaseFormField label="Issued Date">
              <BaseTextInput
                name="issuedOn"
                placeholder="Enter the issue date for the credentials"
                onChange={handleChanges}
                value={credsInputs.issuedOn}
                required
              />
            </BaseFormField>
            <BaseFormField label="Expiration Date">
              <BaseTextInput
                name="expirationDate"
                placeholder="Enter the expiration date for the credentials"
                onChange={handleChanges}
                value={credsInputs.expirationDate}
              />
            </BaseFormField>
            <BaseButton
              margin="medium"
              type="submit"
              primary
              label="Submit"
              alignSelf="center"
            />
          </Box>
        </BaseForm>
      </section>
    </Container>
  );
};

CredentialsForm.propTypes = {
  history: PropTypes.object.isRequired // eslint-disable-line react/forbid-prop-types
};

const Container = styled.main`
  width: 100%;
  height: calc(100vh - 70px);
  padding: 120px 3% 0;
  position: relative;

  section {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    width: 375px;
    height: 100vh;
    background: ${props => props.theme.global.colors.dashBoardBg};
    padding: 120px 3% 0;
    border-left: 1px solid ${props => props.theme.global.colors.dashBoardBorder};
    overflow-x: hidden;
    overflow-y: auto;

    h2 {
      width: 100%;
      text-align: center;
      margin-bottom: 37px;
    }
    form {
      width: 100%;
      display: flex;
      flex-direction: column;
    }
  }
`;

export default CredentialsForm;
