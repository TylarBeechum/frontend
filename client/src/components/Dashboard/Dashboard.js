import React from 'react';
import NewSchoolForm from '../Onboarding/OnboardForm';

const Dashboard = () => {
  return (
    <div>
      <NewSchoolForm />
    </div>
  );
};

export default Dashboard;
