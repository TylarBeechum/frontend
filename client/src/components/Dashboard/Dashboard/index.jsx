import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { useStateValue } from 'react-conflux';

import SchoolDashboard from './SchoolDashboard';
import StudentDashboard from './StudentDashboard';
import queries from './queries';
import {
  schoolContext,
  SCHOOL_DATA_START,
  SCHOOL_DATA_SUCCESS,
  SCHOOL_DATA_ERROR
} from '../../../store/reducers/schoolReducer';
import {
  studentContext,
  STUDENT_DATA_START,
  STUDENT_DATA_SUCCESS,
  STUDENT_DATA_ERROR
} from '../../../store/reducers/studentReducer';
import { globalContext } from '../../../store/reducers/globalReducer';

const Dashboard = ({ history }) => {
  const [{ user }] = useStateValue(globalContext);
  const [schoolState, schoolDispatch] = useStateValue(schoolContext);
  const [, studentDispatch] = useStateValue(studentContext);
  useEffect(() => {
    if (!schoolState.schoolData) {
      user.roleId === '2'
        ? schoolDispatch({ type: SCHOOL_DATA_START })
        : studentDispatch({ type: STUDENT_DATA_START });
      async function getUserData() {
        try {
          const { id } = user;
          const data = await queries.getUserById({
            id
          });
          user.roleId === '2'
            ? schoolDispatch({ type: SCHOOL_DATA_SUCCESS, payload: data })
            : studentDispatch({ type: STUDENT_DATA_SUCCESS, payload: data });
        } catch (err) {
          user.roleId === '2'
            ? schoolDispatch({ type: SCHOOL_DATA_ERROR })
            : studentDispatch({ type: STUDENT_DATA_ERROR });
        }
      }
      getUserData();
    }
  }, [schoolDispatch, schoolState.schoolData, user, studentDispatch]);
  return (
    <Container>
      {user.roleId === '2' ? (
        <SchoolDashboard history={history} />
      ) : (
        <StudentDashboard history={history} />
      )}
    </Container>
  );
};

SchoolDashboard.propTypes = {
  history: PropTypes.object.isRequired // eslint-disable-line react/forbid-prop-types
};

const Container = styled.div`
  padding: 70px 2% 0;
  margin: 0 auto;
  min-height: calc(100vh - 70px);
  width: 100%;
  background: #f8f8f8;
`;

export default Dashboard;
