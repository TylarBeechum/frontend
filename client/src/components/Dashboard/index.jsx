import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { useAuth0 } from '../../auth/authWrapper';

const Dashboard = ({ history }) => {
  const { getTokenWithPopup } = useAuth0();
  const callApi = async () => {
    try {
      await getTokenWithPopup({
        audience: `${process.env.REACT_APP_AUTH0_AUDIENCE}`,
        scope: 'openid profile email offline_access'
      });
    } catch (error) {
      console.error(error);
    }
  };
  return (
    <Container>
      <h1>Dashboard</h1>
      <button type="button" onClick={callApi}>
        Check if working
      </button>
      <button type="button" onClick={() => history.push('/dashboard/credForm')}>
        Issue Credential
      </button>
    </Container>
  );
};

Dashboard.propTypes = {
  history: PropTypes.object.isRequired // eslint-disable-line react/forbid-prop-types
};

const Container = styled.div`
  width: 100%;
  min-height: 100vh;
`;

export default Dashboard;
