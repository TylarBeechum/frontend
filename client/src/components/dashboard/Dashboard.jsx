import React from 'react';
import NewSchoolForm from './onboarding/OnboardForm';

const Dashboard = () => {
  return (
    <div>
      <NewSchoolForm />
    </div>
  );
};

export default Dashboard;
