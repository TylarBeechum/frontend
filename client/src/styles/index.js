import { createGlobalStyle } from 'styled-components';

import Reset from './Reset';
import Global from './Global';

const GlobalStyle = createGlobalStyle`
  ${Reset}
  
  ${Global}

  background-color: red;
`;

export default GlobalStyle;
