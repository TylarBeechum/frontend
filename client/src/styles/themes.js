import styled from 'styled-components';
import { Button, Form, TextInput, FormField } from 'grommet';

export const theme = {
  global: {
    colors: {
      // primary: '#444444',
      // secondary: 'palevioletred',
      fontColor: '#333333',
      bodyBackground: '#f8f8f8',
      teal: 'accent-1',
      dashBoardBorder: '#adadad',
      dashBoardBg: '#ffffff',
      btnHoverColor: '#81FCED',
      focus: 'accent-4',
      navbarHoverBg: '#ffffff',
      searchBarBorder: '#adadad',
      searchBarColor: '#adadad',
      dashboardNotFoundColor: '#adadad',
      mobileTrashBtnBorder: '#adadad',
      confirmationInputBorder: '#adadad',
      footerColor: '#adadad'
    },
    border: '1px solid rgba(0,0,0,0.33)',
    // fontFamily
    radius: '6px',
    font: {
      family: 'Roboto'
    }
  },
  tab: {
    active: {
      color: 'brand'
    },
    color: '#333333',
    border: {
      color: '#f8f8f8',
      active: {
        color: 'brand'
      }
    }
  }
};

export const BaseButton = styled(Button)``;

export const SecondaryButton = styled(Button)`
  color: white;
  border: 1px solid white;
  :hover {
    border: 2px solid white;
  }
`;

export const BaseForm = styled(Form)``;

export const BaseTextInput = styled(TextInput)``;

export const BaseFormField = styled(FormField)``;
