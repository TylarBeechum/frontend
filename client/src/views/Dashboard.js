import React from 'react';
import NewSchoolForm from '../components/OnBoarding.jsx/NewSchoolForm';

const Dashboard = () => {
  return (
    <div>
      <NewSchoolForm />
    </div>
  );
};

export default Dashboard;
