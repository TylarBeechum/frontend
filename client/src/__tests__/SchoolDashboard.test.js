/* eslint-disable */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { Grommet } from 'grommet';
import { theme } from '../styles/themes';

import {
  renderWithRouterAndProviders,
  rendererWithRouterAndProviders
} from '../testHelpers';
import Dashboard from '../components/Dashboard/SchoolDashboard';

describe('<Dashboard />', () => {
  describe('tests that <Dashboard /> renders without crashing', () => {
    it('should renders the SchoolDashboard component without crashing', () => {
      const helpers = renderWithRouterAndProviders(
        <Grommet theme={theme}>
          <Dashboard history={{ location: { pathname: '/' } }} />
        </Grommet>
      );
    });
  });

  describe('tests <Dashboard /> matches previous snapshot', () => {
    it('matches the snapshot of Dashboard', () => {
      const tree = rendererWithRouterAndProviders(
        <Grommet theme={theme}>
          <Dashboard history={{ location: { pathname: '/' } }} />
        </Grommet>
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
