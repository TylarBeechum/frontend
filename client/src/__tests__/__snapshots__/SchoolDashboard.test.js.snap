// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SchoolDetailsForm /> should match the previous snapshot of the component tests that the component has not changed and matched the previous snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 iDwiuh"
>
  <section
    className="sc-cSHVUG chVKOk"
  >
    <h2>
      testName
    </h2>
    <div>
      <input
        name="searchText"
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
      <button
        className="sc-bdVaJa sc-kAzzGY hWLJXC StyledButton-sc-323bzc-0 aNlbA"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        type="button"
      >
        + Issue Credential
      </button>
    </div>
  </section>
  <div
    className="sc-kgoBCf fzXlUD StyledBox-sc-13pk1d4-0 IRSNj"
  >
    <section
      className="sc-VigVT kwgGCs"
    >
      <div
        className="sc-EHOje hWneJQ"
      >
        <button
          aria-label="view credentials button"
          className="sc-bdVaJa sc-bZQynM csIfUH StyledButton-sc-323bzc-0 kHrfdt"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          View
        </button>
      </div>
      <div
        className="sc-gzVnrw bKsZFP"
      >
        <h2>
          test1
        </h2>
        <p>
          test
        </p>
      </div>
      <div
        className="sc-htoDjs erhmUl"
      >
        <p />
      </div>
      <div
        className="sc-dnqmqq jubHKb"
      >
        <h3>
          Nathan Thomas
        </h3>
      </div>
    </section>
    <section
      className="sc-VigVT kwgGCs"
    >
      <div
        className="sc-EHOje hWneJQ"
      >
        <button
          aria-label="view credentials button"
          className="sc-bdVaJa sc-bZQynM csIfUH StyledButton-sc-323bzc-0 kHrfdt"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          View
        </button>
      </div>
      <div
        className="sc-gzVnrw bKsZFP"
      >
        <h2>
          test2
        </h2>
        <p>
          test
        </p>
      </div>
      <div
        className="sc-htoDjs erhmUl"
      >
        <p />
      </div>
      <div
        className="sc-dnqmqq jubHKb"
      >
        <h3>
          Nathan Thomas
        </h3>
      </div>
    </section>
    <section
      className="sc-VigVT kwgGCs"
    >
      <div
        className="sc-EHOje hWneJQ"
      >
        <button
          aria-label="view credentials button"
          className="sc-bdVaJa sc-bZQynM csIfUH StyledButton-sc-323bzc-0 kHrfdt"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          View
        </button>
      </div>
      <div
        className="sc-gzVnrw bKsZFP"
      >
        <h2>
          test3
        </h2>
        <p>
          test
        </p>
      </div>
      <div
        className="sc-htoDjs erhmUl"
      >
        <p />
      </div>
      <div
        className="sc-dnqmqq jubHKb"
      >
        <h3>
          Nathan Thomas
        </h3>
      </div>
    </section>
  </div>
</div>
`;
